src/ggml.c

```diff
@@ -19147,10 +19318,29 @@ void ggml_graph_reset(struct ggml_cgraph * cgraph) {
     GGML_ASSERT(cgraph->grads != NULL);
 
     for (int i = 0; i < cgraph->n_nodes; i++) {
+        struct ggml_tensor * node = cgraph->nodes[i];
         struct ggml_tensor * grad = cgraph->grads[i];
 
+        // initial gradients of loss should be 1, 0 otherwise
         if (grad) {
-            ggml_set_zero(grad);
+            if (node->flags & GGML_TENSOR_FLAG_LOSS) {
```

Comment:
Would it make sense to add a sanity check that only one node is of type `LOSS` to prevent misuse of the API? (unless there are situations where multiple `LOSS` nodes are valid)

Reply:
If you specify multiple loss nodes the result would be that the effective loss would be the sum of individual loss tensors. I wouldn't consider this a misuse of the API since it would be a relatively convenient way to add extra terms to the loss (e.g. l1/l2 regularization).